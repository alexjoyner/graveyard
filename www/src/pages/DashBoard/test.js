import React from 'react';
import { shallow } from 'enzyme';
import { DashBoard } from '.';

describe('DashBoard Page', () => {
  let component;
  beforeEach(() => {
    component = shallow(<DashBoard />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should have a SideBar', () => {
    expect(component.find('SideBar')).toHaveLength(1);
  });
  it('Should contain a Header', () => {
    expect(component.find('Header')).toHaveLength(1);
  });
  describe('HeaderSideBarPageBody', () => {
    let HeaderSideBarPageBody;
    beforeEach(() => {
      HeaderSideBarPageBody = component.find('HeaderSideBarPageBody');
    });
    it('Should have one instance', () => {
      expect(HeaderSideBarPageBody).toHaveLength(1);
    });
    it('Should contain a DashBody', () => {
      expect(HeaderSideBarPageBody.find('Connect(TDashBody)')).toHaveLength(1);
    });
  });
  it('Should have ChartModal', () => {
    expect(component.find('Connect(TChartModal)')).toHaveLength(1);
  }); 
});
