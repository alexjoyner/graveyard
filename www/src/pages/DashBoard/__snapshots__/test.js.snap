// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard Page Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashBoard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-kkGfuU",
          "isStatic": false,
          "rules": Array [
            "
  margin-left: ",
            [Function],
            ";
  min-height: 10px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-kkGfuU",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "SideBar": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-kEYyzF",
          "isStatic": true,
          "rules": Array [
            "
  position: fixed;
  z-index: ",
            "1",
            ";
  background-color: ",
            "#b2bec3",
            ";
  height: 100vh;
  width: 300px;
  left: 0px;
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "render": [Function],
        "styledComponentId": "sc-kEYyzF",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "Body": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-kkGfuU",
            "isStatic": false,
            "rules": Array [
              "
  margin-left: ",
              [Function],
              ";
  min-height: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-kkGfuU",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "SideBar": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-kEYyzF",
            "isStatic": true,
            "rules": Array [
              "
  position: fixed;
  z-index: ",
              "1",
              ";
  background-color: ",
              "#b2bec3",
              ";
  height: 100vh;
  width: 300px;
  left: 0px;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "render": [Function],
          "styledComponentId": "sc-kEYyzF",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
